casepro/statistics/migrations/0001_initial.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+from django.conf import settings
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('orgs', '0016_taskstate_is_disabled'),
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ('cases', '0037_partner_is_restricted'),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='DailyOrgCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('org', models.ForeignKey(to='orgs.Org')),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='DailyUserCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('org', models.ForeignKey(to='orgs.Org')),
+                ('user', models.ForeignKey(to=settings.AUTH_USER_MODEL)),
+            ],
+            options={
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='DailyPartnerCount',
+            fields=[
+                ('id', models.AutoField(verbose_name='ID', serialize=False, auto_created=True, primary_key=True)),
+                ('type', models.CharField(max_length=1)),
+                ('day', models.DateField(help_text='The day this count is for')),
+                ('count', models.PositiveIntegerField(default=1)),
+                ('squashed', models.BooleanField(default=False)),
+                ('partner', models.ForeignKey(to='cases.Partner')),
+            ],
```

Comment:
Ok so first attempt at this was keeping everything hierarchical - i.e. don't record explicit counts for orgs, but aggregate over all users. That started to get expensive, and complicated.

Second attempt was what you're describing. But that felt like it was getting messy because we already have a type field which will be used things like REPLIES vs FORWARDS vs INCOMING, but org vs partner vs user (lets call that "level") didn't easily fit into that. I ended up with all this type specific configuration for each level, and some methods only make sense for a single level.

In the end it felt simpler to pull the different "levels" into separate tables, but keep the thing being counted as a type field.